src/components/GoalCriteria/GoalCriteria.tsx

```diff
@@ -298,60 +208,70 @@ const GoalCriteriaItem: React.FC<GoalCriteriaItemProps> = memo((props) => {
-                {nullable(availableActions, (actions) => (
-                    <Dropdown
-                        onChange={handleChange}
-                        renderTrigger={({ onClick }) => <IconMoreVerticalOutline size="xs" onClick={onClick} />}
-                        items={actions}
-                        renderItem={(props) => (
-                            <MenuItem
-                                key={props.index}
-                                onClick={props.onClick}
-                                icon={props.item.icon}
-                                ghost
-                                color={props.item.color}
-                            >
-                                {props.item.label}
-                            </MenuItem>
-                        )}
-                    />
-                ))}
-            </ContentItem>
-        </StyledTableRow>
+        <StyledTableRow
+            actions={availableActions}
+            onActionClick={handleChange}
+            item={{
+                title,
+                weight,
+                issuer,
+                owner,
+                projectId,
+                scopeId,
+                state,
+                estimate,
+            }}
+            rowIcon={!goalAsCriteria && <GoalCriteriaCheckBox onClick={onToggle} checked={isDone} canEdit={canEdit} />}
+            columns={[
+                {
+                    name: 'title',
+                    renderColumn(values) {
+                        if (!goalAsCriteria) {
+                            return (
+                                <CustomCell>
+                                    <Title size="s" weight="thin">
+                                        {values.title}
+                                    </Title>
+                                </CustomCell>
+                            );
+                        }
+
+                        return (
+                            <CustomCell>
+                                <NextLink passHref href={routes.goal(`${values.projectId}-${values.scopeId}`)}>
+                                    <Title size="s" weight="bold">
+                                        {values.title}
+                                    </Title>
+                                </NextLink>
+                            </CustomCell>
+                        );
+                    },
+                },
+                {
+                    name: 'weight',
+                    renderColumn: ({ weight }) => (
+                        <CustomCell align="right">
+                            {nullable(weight, (w) => (
+                                <Text size="s">{w}</Text>
+                            ))}
+                        </CustomCell>
+                    ),
+                },
+                { name: 'state' },
+                { name: 'projectId' },
+                {
+                    name: 'issuers',
+                    renderColumn: (values) => (
```

Comment:
`({ issuers })`

also I really hate render functions

Reply:
well i could rewrite this with React.Context but eventually will get render function in consumer :)